Maven build fails to resolve dependencies in insecure repository when using maven 3.8.1

### Overview

In Maven 3.8.1, a [settings change](https://maven.apache.org/docs/3.8.1/release-notes.html#cve-2021-26291) was made to block repositories that don't use HTTPS.  Since dataloader uses the repository `http://maven-eclipse.github.io/maven`, builds that need to download the project dependencies fail with the following message:

> [ERROR] Failed to execute goal on project dataloader: Could not resolve dependencies for project com.force:dataloader:jar:43.0.0: Failed to collect dependencies at org.eclipse.swt:org.eclipse.swt.gtk.linux.x86_64:jar:4.5.1: Failed to read artifact descriptor for org.eclipse.swt:org.eclipse.swt.gtk.linux.x86_64:jar:4.5.1: Could not transfer artifact org.eclipse.swt:org.eclipse.swt.gtk.linux.x86_64:pom:4.5.1 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for repositories: [maven-eclipse-repo (http://maven-eclipse.github.io/maven, default, releases+snapshots)] -> [Help 1]

### Reproduction

You can reproduce this issue with this Dockerfile:

```
FROM maven:adoptopenjdk
  
ADD https://github.com/forcedotcom/dataloader/archive/V43.0.zip ./

RUN apt update && \
    apt install unzip && \
    unzip V43.0.zip && \
    cd ./dataloader-43.0 && \
    mvn clean package -DskipTests
```

### Version Info

- Dataloader: 43.0
- Maven: 3.8.1
- Java:  OpenJDK 15.0.2

--- a/pom.xml
+++ b/pom.xml
@@ -41,7 +41,7 @@
     </repository>
     <repository>
       <id>maven-eclipse-repo</id>
-      <url>http://maven-eclipse.github.io/maven</url>
+      <url>https://maven-eclipse.github.io/maven</url>
     </repository>
   </repositories>
 

